certificate-manager/src/main/java/io/strimzi/certs/IpAndDnsValidation.java

```diff
@@ -28,7 +28,10 @@ public class IpAndDnsValidation {
                 "[a-zA-Z\\d]|" +
                 // followed by more labels of same
                 "[a-zA-Z\\d][a-zA-Z\\d\\-]{0,61}[a-zA-Z\\d])";
-        DNS_NAME = Pattern.compile("^" + dnsLabel + "(\\." + dnsLabel + ")*");
+        // a trailing dot that can be used to force FQDN resolutions, reducing
+        // the internal DNS server load (kube-dns pods) and application latency
+        String trailingDot = "(\\.?)";
```

Comment:
I guess we don't need the parens, because we don't want to capture the match for `.?`, right?

Reply:
Yep, removed.